src/minter-suite-mapping.ts

```diff
@@ -913,281 +940,242 @@ export function handleProjectNextTokenEjected(
 // Currently supported key-value types (value: T) include boolean, BigInt, ETH address, and bytes values.
 // For any questions reach out to @jon or @ryley-o.eth. or see the following document https://docs.google.com/document/d/1XSxl04eJyTxc_rbj6cmq-j00zaYDzApBBLT67JXtaOw/edit?disco=AAAAZa8xp-Q
 
-export function handleSetMinterDetailsGeneric<T, C>(
+export function setProjectMinterConfigExtraMinterDetailsValue<ValueType>(
   key: string,
-  value: T,
-  config: C
+  value: ValueType,
+  config: ProjectMinterConfiguration
 ): void {
-  if (
-    !(config instanceof Minter || config instanceof ProjectMinterConfiguration)
-  ) {
-    log.warning(
-      "[WARN] Generic property attempted to be set on something not a Minter or ProjectMinterConfiguration",
-      []
-    );
-    return;
-  }
-  let minterDetails = getMinterDetails(config);
-  let jsonKey = key;
-  let jsonValue: JSONValue;
-
-  if (isBoolean(value)) {
-    jsonValue = json.fromString(booleanToString(value));
-  } else if (value instanceof BigInt) {
-    jsonValue = json.fromString(value.toString());
-  } else if (value instanceof Address) {
-    jsonValue = stringToJSONValue(value.toHexString());
-  } else if (value instanceof Bytes) {
-    jsonValue = bytesToJSONValue(value);
-  } else {
-    log.warning(
-      "handleSetMinterDetailsGeneric received unexpected typed value",
-      []
-    );
-    return;
-  }
+  let minterDetails = getProjectMinterConfigExtraMinterDetails(config);
+
+  minterDetails = createUpdatedTypedMapWithEntryAdded(
+    minterDetails,
+    key,
+    value
+  );
 
-  minterDetails.set(jsonKey, jsonValue);
+  // minterDetails.set(jsonKey, jsonValue);
   config.extraMinterDetails = typedMapToJSONString(minterDetails);
   config.save();
 }
 
-export function handleSetValueProjectConfig<T>(event: T): void {
-  if (
-    !(
-      event instanceof ConfigValueSetBool ||
-      event instanceof ConfigValueSetBigInt ||
-      event instanceof ConfigValueSetAddress ||
-      event instanceof ConfigValueSetBytes
-    )
-  ) {
-    return;
-  }
-  let minterProjectAndConfig = loadMinterProjectAndConfig(
-    event.address,
-    event.params._projectId,
-    event.block.timestamp
+export function setMinterExtraMinterDetailsValue<ValueType>(
+  key: string,
+  value: ValueType,
+  minter: Minter
+): void {
+  let minterDetails = getMinterExtraMinterDetailsTypedMap(minter);
+
+  minterDetails = createUpdatedTypedMapWithEntryAdded(
+    minterDetails,
+    key,
+    value
   );
-  if (minterProjectAndConfig) {
-    const config = minterProjectAndConfig.projectMinterConfiguration;
-    if (
-      !(
-        config instanceof Minter || config instanceof ProjectMinterConfiguration
-      )
-    ) {
-      return;
-    }
 
-    handleSetMinterDetailsGeneric(
-      event.params._key.toString(),
-      event.params._value,
-      config
-    );
+  minter.extraMinterDetails = typedMapToJSONString(minterDetails);
+  minter.save();
```

Comment:
There may be opportunity to include an `updatedAt` bump here, but regardless, we already handle updating that outside of the helper function this is replacing, and probably better to do outside of this PR (if ever)

Reply:
Good point, I'll just add now, not too many places that need updating!

Reply:
sounds good to me, resolving this 🙏